script/run-unit-test.js

```diff
@@ -0,0 +1,37 @@
+const printUnitTestHelp = require('./run-unit-test-help.js');
+const commandLineArgs = require('command-line-args');
+const { runCommand } = require('./utils');
+
+const COMMAND_LINE_OPTIONS_DEFINITIONS = [
+  { name: 'log-level', type: String, defaultValue: 'log' },
+  { name: 'coverage', type: Boolean, defaultValue: false },
+  { name: 'help', alias: 'h', type: Boolean, defaultValue: false },
+  {
+    name: 'path',
+    type: String,
+    defaultOption: true,
+    defaultValue: './src/**/*.unit.spec.js?(x)',
+  },
+];
+const options = commandLineArgs(COMMAND_LINE_OPTIONS_DEFINITIONS);
+
+if (options.help) {
+  printUnitTestHelp();
+  process.exit(0);
+}
+
+const mochaPath = 'BABEL_ENV=test mocha';
+const coveragePath =
+  'NODE_ENV=test nyc --all --reporter=lcov --reporter=text-summary mocha --reporter mocha-junit-reporter --no-color';
+const testRunner = options.coverage ? coveragePath : mochaPath;
+const mochaOpts =
+  'src/platform/testing/unit/mocha.opts src/platform/testing/unit/helper.js';
```

Comment:
Yeah, my hope would be that we could combine them into one file, since that's easier overall. But I didn't try to switch Jenkins yet.

I don't think `--all` needs to be conditional on having provided path. It basically sets the scope of files to check for coverage, and when you specify a path, that overall coverage numbers are misleading no matter what.